geopandas/base.py

```diff
@@ -135,6 +137,24 @@ def is_ring(self):
         return Series([geom.exterior.is_ring for geom in self.geometry],
                       index=self.index)
 
+    @property
+    def x(self):
+        """Return the x location of point geometries in a GeoSeries"""
+        xloc = _series_unary_op(self, 'x', null_value=np.nan)
+        if not xloc.isnull().all():
+            return xloc
+        else:
+            raise ValueError("No geom_type with x attribute in GeoSeries")
+
+    @property
+    def y(self):
+        """Return the y location of point geometries in a GeoSeries"""
+        yloc = _series_unary_op(self, 'y', null_value=np.nan)
+        if not yloc.isnull().all():
+            return yloc
+        else:
+            raise ValueError("No geom_type with y attribute in GeoSeries")
```

Comment:
I was wondering whether this would be a problem that this raises based on the return values, and not based on the actual types. 
But probably this is not a problem? It would only be a problem in the case where you have `Point`s with NaN coordinates, or when having an empty series (eg due to a filtering operation), but for an empty we would also not be able to infer that it is an empty 'Point GeoSeries')